sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/TooManyGenericParameters.cs

```diff
@@ -41,10 +41,14 @@ public class TooManyGenericParameters : ParameterLoadingDiagnosticAnalyzer
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(rule);
 
         private const int DefaultValueMaximum = 2;
-
         [RuleParameter("max", PropertyType.Integer, "Maximum authorized number of generic parameters.", DefaultValueMaximum)]
         public int Maximum { get; set; } = DefaultValueMaximum;
 
+        private const int DefaultValueMaximumMethod = 3;
+        [RuleParameter("maxMethod", PropertyType.Integer, "Maximum authorized number of generic parameters for methods.",
+            DefaultValueMaximumMethod)]
+        public int MaximumMethod { get; set; } = DefaultValueMaximumMethod;

```

Comment:
The name is a confusing. Don't have good suggestion though. Maybe MaximumTypeParametersInMethod?